### ISSUE TITLE
Template render error could use more information

### ISSUE THREAD

**Issue by mbforbes** [11ty/eleventy#2216]
Hi there,

First off, thank you for building Eleventy, I'm very excited to try using it.

I'm migrating an old Jekyll site with lots of Liquid templates to Eleventy + Nunjuks. I'm running into lots of template errors during my migration.

The issue I'm having is that the error messages I get make it very hard to find the source of my problems in a big (couple hundred line) template file. The complete error message is something like:

```txt
[11ty] (./_includes/layouts/default.njk)
  Error: template names must be a string: undefined
```

It would be so helpful if it gave me any more information—e.g., even the line that it failed on would be a big help. Running with `DEBUG=* ...` prints a lot more info about Eleventy's code, but I can't see any more info about the template.

Many thanks for considering this, and apologies if it's already suggested — I tried an issue search but couldn't see a duplicate at first glance.

**Comment by pdehaan** [comment]
Yeah, I think we've had a few existing reports on error messaging in the past, which I'll have to try and dig up and cross reference.

11ty natively supports liquidjs though. Curious how well an old Jekyll site could be converted to 11ty (without switching to Nunjucks) or what issues come up between the reference Ruby liquid implementation and the JavaScript liquidjs implementation. I know https://github.com/harttle/liquidjs has been doing some great work on adding some Jekyll compatibility to recent liquidjs 9.x builds (which should be available if you're using Eleventy v1.0.0+).

**Comment by zachleat** [comment]
I don’t think `template names must be a string: undefined` is an Eleventy error, it might be from a dependency. Did you figure out what this specific issue was? Might help improve the error messaging overall

**Comment by mbforbes** [comment]
I think it was an error bubbled up from Nunjucks, and the issue was me using `{% include foo.njk %}` instead of `{% include "foo.njk" %}`.

I wonder whether Nunjucks includes the line causing the error somewhere in the reporting, but not in the error message string, so it gets swallowed when only the message is surfaced? Or could be their error has no more info at all, and I'm barking up the wrong tree 😅
